File: programmers/81302/81302.js
// 2021 카카오 채용 연계형 인턴십
// 거리두기 확인하기

function find(place) {
  const SIZE = 5;
  for (const [i, row] of place.entries()) {
    for (let j = 0; j < row.length; j++) {
      const dot = row.charAt(j);
      let p_array = [];
      if (dot === "O" || dot === "P") {
        if (i > 0) p_array = [...p_array, place[i - 1][j]];
        if (i < SIZE - 1) p_array = [...p_array, place[i + 1][j]];
        if (j > 0) p_array = [...p_array, place[i][j - 1]];
        if (j < SIZE - 1) p_array = [...p_array, place[i][j + 1]];
        const p_count = p_array.filter((element) => element === "P").length;
        if ((dot === "O" && p_count >= 2) || (dot === "P" && p_count >= 1)) {
          return false;
        }
      }
    }
  }
  return true;
}
function solution(places) {
  let answer = [];
  places.forEach((place) => {
    answer = [...answer, find(place) ? 1 : 0];
  });
  return answer;
}

File: programmers/81302/81302.py
# 2021 카카오 채용 연계형 인턴십
# 거리두기 확인하기

def find(place):
    SIZE = 5
    for i, row in enumerate(place):
        for j, dot in enumerate(row):
            p_array = []
            if dot == "O" or dot == "P":
                if (i > 0):
                    p_array.append(place[i-1][j])
                if (i < SIZE - 1):
                    p_array.append(place[i+1][j])
                if (j > 0):
                    p_array.append(place[i][j-1])
                if (j < SIZE - 1):
                    p_array.append(place[i][j+1])
                p_count = p_array.count("P")
                if (dot == "O" and p_count >= 2) or (dot == "P" and p_count >= 1):
                    return False
    return True


def solution(places):
    answer = []
    for i, place in enumerate(places):
        print(find(place))
        answer.append(1 if find(place) else 0)

    return answer
